lib/OpenQA/Utils.pm

```diff
@@ -224,6 +224,27 @@ sub needle_info {
     $needle->{name}      = $name;
     $needle->{distri}    = $distri;
     $needle->{version}   = $version;
+
+	#After change the needle's properties form to hash, support backward compatibility
+	return $needle unless @{$needle->{properties}};
+
+	my @properties = ();
+	foreach (@{$needle->{properties}}){
+		if (ref($_) eq "HASH"){
+			push (@properties, $_);
+			next;
+		}
+		if ($_ eq "workaround"){
```

Comment:
In my opinion you're carrying the use of `$_` to far here. Use a variable with a decent name.

Reply:
I will use $property to instead of $_